yaml/src/Data/Yaml.hs

```diff
@@ -188,16 +189,20 @@ encodeFile = encodeFileWith defaultEncodeOptions
 -- @since 0.10.2.0
 encodeFileWith :: ToJSON a => EncodeOptions -> FilePath -> a -> IO ()
 encodeFileWith opts fp obj = runConduitRes
-    $ CL.sourceList (objToEvents opts $ toJSON obj)
+    $ CL.sourceList (objToStream opts obj)
    .| Y.encodeFileWith (encodeOptionsFormat opts) fp
 
-objToEvents :: EncodeOptions -> Value -> [Y.Event]
-objToEvents opts o = (:) EventStreamStart
+objToStream :: ToJSON a => EncodeOptions -> a -> [Y.Event]
+objToStream opts o = (:) EventStreamStart
               . (:) EventDocumentStart
-              $ objToEvents' o
+              $ objToEvents opts o ++
               [ EventDocumentEnd
               , EventStreamEnd
               ]
+
+-- | Encode a value into YAML 'Event's
```

Comment:
Please add a `@since` Haddock comment, as well as a minor version bump and ChangeLog entry.